packages/chat-stateful-client/src/ChatContext.ts

```diff
@@ -50,21 +50,15 @@ export class ChatContext {
     return this._state;
   }
 
-  public setState(state: ChatClientState): void {
-    this._state = state;
-    if (!this._batchMode) {
-      this._emitter.emit('stateChanged', this._state);
-    }
-  }
-
   public modifyState(modifier: (draft: ChatClientState) => void): void {
+    const priorState = this._state;
     this._state = produce(this._state, modifier, (patches: Patch[]) => {
       if (getLogLevel() === 'verbose') {
         // Log to `info` because AzureLogger.verbose() doesn't show up in console.
         this._logger.info(`State change: ${JSON.stringify(patches)}`);
       }
     });
-    if (!this._batchMode) {
+    if (!this._batchMode && this._state !== priorState) {
```

Comment:
Good to see we are adding one more checker to this
Curious about if there is a case that someone calls modifyState but state wouldn't change - Is there an existing case for this? Maybe like we set a string, but it is the same value as the existing one?